src/meta/api/src/data_mask_api_impl.rs

```diff
@@ -0,0 +1,227 @@
+// Copyright 2021 Datafuse Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt::Display;
+
+use common_meta_app::app_error::AppError;
+use common_meta_app::app_error::DatamaskAlreadyExists;
+use common_meta_app::app_error::TxnRetryMaxTimes;
+use common_meta_app::app_error::UnknownDatamask;
+use common_meta_app::data_mask::CreateDatamaskReply;
+use common_meta_app::data_mask::CreateDatamaskReq;
+use common_meta_app::data_mask::DatamaskId;
+use common_meta_app::data_mask::DatamaskMeta;
+use common_meta_app::data_mask::DatamaskNameIdent;
+use common_meta_app::data_mask::DropDatamaskReply;
+use common_meta_app::data_mask::DropDatamaskReq;
+use common_meta_app::data_mask::GetDatamaskReply;
+use common_meta_app::data_mask::GetDatamaskReq;
+use common_meta_kvapi::kvapi;
+use common_meta_types::ConditionResult::Eq;
+use common_meta_types::MetaError;
+use common_meta_types::TxnRequest;
+use common_tracing::func_name;
+use tracing::debug;
+
+use crate::data_mask_api::DatamaskApi;
+use crate::fetch_id;
+use crate::get_pb_value;
+use crate::get_u64_value;
+use crate::id_generator::IdGenerator;
+use crate::kv_app_error::KVAppError;
+use crate::send_txn;
+use crate::serialize_struct;
+use crate::serialize_u64;
+use crate::txn_cond_seq;
+use crate::txn_op_del;
+use crate::txn_op_put;
+use crate::TXN_MAX_RETRY_TIMES;
+
+/// DatamaskApi is implemented upon kvapi::KVApi.
+/// Thus every type that impl kvapi::KVApi impls DatamaskApi.
+#[tonic::async_trait]
+impl<KV: kvapi::KVApi<Error = MetaError>> DatamaskApi for KV {
+    async fn create_data_mask(
+        &self,
+        req: CreateDatamaskReq,
+    ) -> Result<CreateDatamaskReply, KVAppError> {
+        debug!(req = debug(&req), "DatamaskApi: {}", func_name!());
+
+        let name_key = &req.name;
+
+        let mut retry = 0;
+        while retry < TXN_MAX_RETRY_TIMES {
+            retry += 1;
```

Comment:
Replace it with the following, make error handling cleaner and simpler:
https://github.com/datafuselabs/databend/blob/1b2e4c841f1177c934180d96e3cba4fd529beeb6/src/meta/api/src/schema_api_impl.rs#L2161-L2163